packages/core/src/guards.ts

```diff
@@ -0,0 +1,35 @@
+import { EventObject, StateValue, GuardPredicate } from './types';
+import { isStateId } from './stateUtils';
+import { isString } from 'util';
+
+export function stateIn<TContext, TEvent extends EventObject>(
+  stateValue: StateValue
+): GuardPredicate<TContext, TEvent> {
+  return {
+    type: 'xstate.guard',
+    name: 'In',
```

Comment:
`In()` should be available to all conditional expressions and that includes `<if>` & `<elseif>` represented by `choose`, would be great to add support for this and some basic tests

Reply:
Note: previously `in:` was working with some relative paths, this is not. I've raised concerns about `in:` in the past though because the given string was checked "by relative grandparent" which was different than relative targets for transitions and this was IMHO quite confusing.

Reply:
Yeah, let's deprecate "by relative grandparent" in a separate PR.

Reply:
See card: https://github.com/davidkpiano/xstate/projects/1#card-40695477